cloudinit/util.py

```diff
@@ -1802,6 +1801,24 @@ def time_rfc2822():
         ts = "??"
     return ts
 
+def boottime():
+    import ctypes
+
+    NULL_BYTES = b"\x00"
+
+    class timeval(ctypes.Structure):
+        _fields_ = [
+            ("tv_sec", ctypes.c_int64),
+            ("tv_usec", ctypes.c_int64)
+        ]
+    libc = ctypes.CDLL('/lib/libc.so.7')
+    size = ctypes.c_size_t()
```

Comment:
I think we want to use an instance of the timeval class.

buf = timeval()
size = ctypes.c_size_t()
size.value = ctypes.sizeof(buf)

Reply:
we, correctly, get a size of 16 with the current method, and should get the same with your method